tests/test_views_item_put.py

```diff
@@ -83,26 +88,37 @@ def test_valid_put_etag(app, test_records, content_type):
                 'If-Match': '"{0}"'.format(record.revision_id)
             })
         assert res.status_code == 200
-
         assert get_json(client.get(url))['metadata']['year'] == 1234
 
+        IndexFlusher(search_class).flush_and_wait()
+        res = client.get(search_url, query_string={"year": 1234})
+        assert_hits_len(res, 1)
+
 
 @pytest.mark.parametrize('content_type', [
     'application/json', 'application/json;charset=utf-8'
 ])
-def test_put_on_deleted(app, test_records, content_type):
+def test_put_on_deleted(app, db, es, test_data, content_type, search_url,
+                        search_class):
     """Test putting to a deleted record."""
-    HEADERS = [
-        ('Accept', 'application/json'),
-        ('Content-Type', content_type)
-    ]
-
-    # create the record using the internal API
-    pid, record = test_records[0]
-
     with app.test_client() as client:
-        url = record_url(pid)
+        HEADERS = [
+            ('Accept', 'application/json'),
+            ('Content-Type', content_type)
+        ]
+        HEADERS.append(('Content-Type', content_type))
+
+        # Create record
+        res = client.post(
```

Comment:
why are we not using the `indexed_records` fixture?

Reply:
We would have 3 indexed records remaining and one of them could be returned when we search for it after deleting. It's also cleaner to have 1 record, delete it and then have 0.